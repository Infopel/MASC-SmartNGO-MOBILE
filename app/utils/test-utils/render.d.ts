export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
