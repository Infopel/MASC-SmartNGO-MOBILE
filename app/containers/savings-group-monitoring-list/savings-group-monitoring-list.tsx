import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { Icon } from "components/icon/icon"
import { SavingsGroupMonitoringItem } from "components/savings-group-monitoring-item/savings-group-monitoring-item"
import { translate } from "i18n/translate"
import { observer } from "mobx-react-lite"
import { Fab, FlatList, VStack } from "native-base"
import { navigate } from "navigators/navigation-utilities"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { ISavingsGroupMonitoringItem, listAllSavingsGroupMonitoring } from "storage/queries/savings-group"

export interface SavingsGroupMonitoringListProps {
  /**
   * An optional style override useful for padding & margin.
   */
  savingsGroupId: string
  style?: StyleProp<ViewStyle>
  items: ISavingsGroupMonitoringItem[]
}

/**
 * Describe your component here
 */
const Container = observer(function SavingsGroupMonitoringList(props: SavingsGroupMonitoringListProps) {
  const {  items, savingsGroupId } = props

  return (
    <VStack flex='1'>
      <FlatList data={items} renderItem={({ item }) => <SavingsGroupMonitoringItem item={item} />} />
      <Fab renderInPortal={false} leftIcon={<Icon name='add-circle' size={'xl'} />} label={translate('savings-group-details-monitoring-tab.add-monitoring')} onPress={() => navigate('form', { formId: 'savings-group-monitoring', contextId: savingsGroupId, parentContextId: savingsGroupId })} />
    </VStack>
  )
})

export const SavingsGroupMonitoringListContainer = withDatabase(withObservables(['database', "savingsGroupId"], ({ database, savingsGroupId }) => ({ items: listAllSavingsGroupMonitoring(database, savingsGroupId) }))(Container))
