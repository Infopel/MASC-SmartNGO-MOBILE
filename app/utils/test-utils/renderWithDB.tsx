jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  