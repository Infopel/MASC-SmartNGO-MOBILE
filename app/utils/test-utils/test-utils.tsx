import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

