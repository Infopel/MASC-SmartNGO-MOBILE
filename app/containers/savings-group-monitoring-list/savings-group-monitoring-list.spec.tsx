import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { SavingsGroupMonitoringListContainer } from "./savings-group-monitoring-list"

describe("<SavingsGroupMonitoringListContainer>",()=>{
 it("render", async function () {
    const component = <SavingsGroupMonitoringListContainer></SavingsGroupMonitoringListContainer>
    
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
  