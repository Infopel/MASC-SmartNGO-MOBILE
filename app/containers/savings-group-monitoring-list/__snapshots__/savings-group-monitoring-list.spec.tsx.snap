// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavingsGroupMonitoringListContainer> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    data={
      Array [
        Object {
          "age": 22,
          "id": "ewqdf",
          "name": "Paulo Artur",
          "savings": 50000,
        },
      ]
    }
    dataSet={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": "#fafaf9",
              "borderBottomColor": "#fafafa",
              "borderLeftColor": "#ef8812",
              "borderLeftWidth": 6,
              "borderRadius": 6,
              "borderRightColor": "#fafafa",
              "borderTopColor": "#fafafa",
              "borderWidth": 1,
              "elevation": 2,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 1.41,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Paulo Artur
              </Text>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 8,
                  }
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                22
                 anos
              </Text>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 8,
                  }
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Poupan√ßa: 
                50000
              </Text>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "width": 8,
                }
              }
            />
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              colorScheme="danger"
              dataSet={Object {}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    dataSet={Object {}}
    style={Object {}}
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d37b16",
        "borderRadius": 9999,
        "bottom": 16,
        "elevation": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "position": "absolute",
        "right": 16,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "zIndex": 20,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fafafa",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Add Monitoring
        </Text>
      </View>
    </View>
  </View>
</View>
`;
